concourse/scripts/test_diskquota.sh

```diff
@@ -18,6 +18,9 @@ function _main() {
     tar -xzf /home/gpadmin/bin_diskquota/diskquota-*-*.tar.gz -C /usr/local/greenplum-db-devel
     source /home/gpadmin/gpdb_src/gpAux/gpdemo/gpdemo-env.sh
 
+    gpconfig -c shared_preload_libraries -v "diskquota-2.2"
+    gpstop -ar
+
```

Comment:
@red1452, Why this line here isn't work as expected? 

https://github.com/arenadata/diskquota/blob/df260b2d583b6136d1b8fe1c6982d89127f058b2/tests/regress/sql/config.sql#L4

Where does test failed with error without lines above?

Reply:
Without my fixes:
```
+ cmake --build . --target installcheck
Built target install_test_extension
(using postmaster on Unix socket, port 6000)
============== dropping database "contrib_regression" ==============
NOTICE:  database "contrib_regression" does not exist, skipping
DROP DATABASE
============== creating database "contrib_regression" ==============
CREATE DATABASE
ALTER DATABASE
============== installing gp_inject_fault             ==============
CREATE EXTENSION
============== installing diskquota_test              ==============
ERROR:  [diskquota] booting 2.2.0, but diskquota-2.2 not in shared_preload_libraries. abort. (diskquota.c:178)
command failed: "/usr/local/greenplum-db-devel/bin/psql" -X -c "CREATE EXTENSION IF NOT EXISTS \"diskquota_test\"" "contrib_regression"
======================================================================
DIFF FILE: /home/gpadmin/diskquota_artifacts/tests/regress/regression.diffs
======================================================================


gmake[3]: *** [tests/CMakeFiles/regress] Error 1
gmake[2]: *** [tests/CMakeFiles/regress.dir/all] Error 2
gmake[1]: *** [tests/CMakeFiles/installcheck.dir/rule] Error 2
gmake: *** [installcheck] Error 2
[root@5be6c7bf072f gpadmin]# su - gpadmin
Last login: Wed Apr 19 10:07:02 UTC 2023 on pts/0
-bash-4.2$ source /usr/local/greenplum-db-devel/greenplum_path.sh;
-bash-4.2$ source gpdb_src/gpAux/gpdemo/gpdemo-env.sh;
-bash-4.2$ gpconfig -s shared_preload_libraries
Values on all segments are consistent
GUC          : shared_preload_libraries
Master  value: 
Segment value: 
-bash-4.2$ exit
```

It does not work because before execute `config.sql` extension `diskquota_test` is configured. Create extension `disquota_test` requires to create extension `diskquota`:
```
postgres=# CREATE EXTENSION IF NOT EXISTS diskquota_test;
ERROR:  [diskquota] booting 2.2.0, but diskquota-2.2 not in shared_preload_libraries. abort. (diskquota.c:178)
```

Loading extension `diskquota_test` is set at `CMakeLists.txt` for `RegressTarget_Add(regress`:
```
RegressTarget_Add(regress
    INIT_FILE
    ${CMAKE_CURRENT_SOURCE_DIR}/init_file
    SQL_DIR ${CMAKE_CURRENT_SOURCE_DIR}/regress/sql
    EXPECTED_DIR ${CMAKE_CURRENT_SOURCE_DIR}/regress/expected${EXPECTED_DIR_SUFFIX}
    RESULTS_DIR ${CMAKE_CURRENT_SOURCE_DIR}/regress/results
    DATA_DIR ${CMAKE_CURRENT_SOURCE_DIR}/data
    SCHEDULE_FILE ${CMAKE_CURRENT_SOURCE_DIR}/regress/diskquota_schedule${EXPECTED_DIR_SUFFIX}
    REGRESS_OPTS
    --load-extension=gp_inject_fault
    --load-extension=diskquota_test
    --dbname=contrib_regression)
```

Reply:
Command `gpconfig -c shared_preload_libraries -v $(./data/current_binary_name);` works correctly:
```
-bash-4.2$ gpconfig -c shared_preload_libraries -v $(./data/current_binary_name);
20230419:10:37:16:012695 gpconfig:5be6c7bf072f:gpadmin-[INFO]:-completed successfully with parameters '-c shared_preload_libraries -v diskquota-2.2.so'
-bash-4.2$ gpstop -raf
20230419:10:37:23:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Starting gpstop with args: -raf
20230419:10:37:23:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Gathering information and validating the environment...
20230419:10:37:23:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Obtaining Greenplum Master catalog information
20230419:10:37:23:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Obtaining Segment details from master...
20230419:10:37:23:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Greenplum Version: 'postgres (Greenplum Database) 6.22.1_arenadata41+dev.307.g60673e0 build commit:60673e0b66bf5af3621a641acef88c9059321870'
20230419:10:37:23:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Commencing Master instance shutdown with mode='fast'
20230419:10:37:23:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Master segment instance directory=/home/gpadmin/gpdb_src/gpAux/gpdemo/datadirs/qddir/demoDataDir-1
20230419:10:37:24:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Attempting forceful termination of any leftover master process
20230419:10:37:24:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Terminating processes for segment /home/gpadmin/gpdb_src/gpAux/gpdemo/datadirs/qddir/demoDataDir-1
20230419:10:37:24:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Stopping master standby host 5be6c7bf072f mode=fast
20230419:10:37:24:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Successfully shutdown standby process on 5be6c7bf072f
20230419:10:37:24:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Targeting dbid [2, 5, 3, 6, 4, 7] for shutdown
20230419:10:37:24:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Commencing parallel primary segment instance shutdown, please wait...
20230419:10:37:24:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-0.00% of jobs completed
20230419:10:37:24:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-100.00% of jobs completed
20230419:10:37:24:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Commencing parallel mirror segment instance shutdown, please wait...
20230419:10:37:24:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-0.00% of jobs completed
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-100.00% of jobs completed
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-----------------------------------------------------
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-   Segments stopped successfully      = 6
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-   Segments with errors during stop   = 0
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-----------------------------------------------------
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Successfully shutdown 6 of 6 segment instances 
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Database successfully shutdown with no errors reported
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Cleaning up leftover gpmmon process
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-No leftover gpmmon process found
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Cleaning up leftover gpsmon processes
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-No leftover gpsmon processes on some hosts. not attempting forceful termination on these hosts
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Cleaning up leftover shared memory
20230419:10:37:25:012833 gpstop:5be6c7bf072f:gpadmin-[INFO]:-Restarting System...
-bash-4.2$ psql postgres
psql (9.4.26)
Type "help" for help.

postgres=# CREATE EXTENSION IF NOT EXISTS diskquota_test;
CREATE EXTENSION
postgres=# create extension diskquota;
CREATE EXTENSION
postgres=# 
```

Reply:
> Where does test failed with error without lines above?

IMHO, command

> --load-extension=diskquota_test

does only 

> CREATE EXTENSION IF NOT EXISTS \"diskquota_test\"

and does it before executing 

> diskquota/tests/regress/sql/config.sql

Reply:
@RekGRpth, @red1452, how creating of diskquota_test leads to creating/initializing diskquota extension? At the first glance, this extension contains only functions definition.

Reply:
As far I understand, there are C-functions in the diskquota_test. Creating of these functions lead to loading shared library and calling PG_init function that throw this the given error. But how it works in the upstream CI?!

Reply:
I have assumption about a possible reason. There is the next line in cmake files: https://github.com/arenadata/diskquota/blob/e4a06ea5a1b8a38eae2c194be11643f48dc8d39e/tests/CMakeLists.txt#L45

I didn't find any guaranties about processing order of dependencies. But if we run isolation tests before regression, all tests pass successfully:

```diff
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 9a77457..904ae8d 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -41,7 +41,7 @@ add_custom_target(install_test_extension

 add_custom_target(installcheck)
 add_dependencies(isolation2 install_test_extension)
-add_dependencies(regress install_test_extension)
+add_dependencies(regress install_test_extension isolation2)
 add_dependencies(installcheck isolation2 regress)
 ```

Anyway, I don't think that it's a better solution...

Reply:
> if we run isolation tests before regression, all tests pass successfully

it is because `isolation2` test has not `--load-extension=diskquota_test` while in `config.sql` it adds `diskquota` to `shared_preload_libraries`